### ISSUE TITLE
Support Pages with dynamic routes (database content, etc.).

### ISSUE THREAD

**Issue by ctolkien** [ctolkien/SodaPop.RazorPagesSitemap#8]
Presumably some type of interface that people would implement which they then register?

**Comment by ctolkien** [comment]
@poke  - any decent ideas on a way to implement this?

Just quickly... perhaps you register your `IRazorSitemapProviderTypeThing` with DI, we pick these up when our services runs, it has a method for `GenerateNode()`, which returns a node... based on.. whatever it wants...?

**Comment by ctolkien** [comment]
```
    public interface IRazorSitemapNodeProvider
    {
        void Register(string path);
        IEnumerable<SitemapNode> Generate();
    }
```

**Comment by ctolkien** [comment]
Resolved in #16
